### PR TITLE
GA - obfuscate ip and user names

### PR DESCRIPTION
**Pull request by cristinamanum**
Add support to anonymizeIp and obfuscate user names in GA.

### CODE REVIEW

File: src/NuGetGallery/App_Code/ViewHelpers.cshtml [hunk 1]
```diff
@@ -353,13 +354,17 @@
...
             {
+		var obfuscatedRequest = ObfuscationHelper.ObfuscateRequestUrl(new HttpContextWrapper(HttpContext.Current), RouteTable.Routes);
                 <script>
                     (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
                     (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
                     m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
                     })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
 
                     ga('create', '@propertyId', 'auto');
+		    ga('set', 'anonymizeIp', true);
```
Comment (shishirx34) [inline src/NuGetGallery/App_Code/ViewHelpers.cshtml]: nit: indentation

File: src/NuGetGallery/Helpers/ObfuscationHelper.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+using System.Web;
+using System.Web.Routing;
+
+namespace NuGetGallery.Helpers
+{
+    public class ObfuscationHelper
+    {
+        public static string ObfuscateRequestUrl(HttpContextBase httpContext, RouteCollection routes)
+        {
+            if (httpContext == null || httpContext.Request == null || httpContext.Request.Url == null || routes == null)
```
Comment (shishirx34) [inline src/NuGetGallery/Helpers/ObfuscationHelper.cs]: Simplify
`if (httpContext?.Request?.Url == null || routes == null)`

File: src/NuGetGallery/Helpers/ObfuscationHelper.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+            {
+                return string.Empty;
+            }
+
+            var route = routes.GetRouteData(httpContext)?.Route as Route;
+            if (route == null)
+            {
+                return string.Empty;
+            }
+            return route.ObfuscateUrlPath(httpContext.Request.Url.AbsolutePath.TrimStart('/'));
```
Comment (scottbommarito) [inline src/NuGetGallery/Helpers/ObfuscationHelper.cs]: nit: use tertiary operator to reduce code

```
var route = routes.GetRouteData(httpContext)?.Route as Route;
return route == null ? string.Empty : route.ObfuscateUrlPath(httpContext.Request.Url.AbsolutePath.TrimStart('/'));
```

File: tests/NuGetGallery.Facts/Helpers/ObfuscationHelperFacts.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+                if (_currentRoutes == null)
+                {
+                    _currentRoutes = new RouteCollection();
+                    Routes.RegisterApiV2Routes(_currentRoutes);
+                    Routes.RegisterUIRoutes(_currentRoutes);
+                }
+            }
+
+            [Fact]
+            public void NullContext()
```
Comment (scottbommarito) [inline tests/NuGetGallery.Facts/Helpers/ObfuscationHelperFacts.cs]: nit: usually test cases have names to describe the expected response in addition to the input

e.g.
```
WithNullContextReturnsEmptyString
```

File: src/NuGetGallery/App_Code/ViewHelpers.cshtml (hunk 1 above)
Comment (cristinamanum) [reply src/NuGetGallery/App_Code/ViewHelpers.cshtml]: Updated

File: src/NuGetGallery/Helpers/ObfuscationHelper.cs (hunk 2 above)
Comment (cristinamanum) [reply src/NuGetGallery/Helpers/ObfuscationHelper.cs]: Updated

File: src/NuGetGallery/Helpers/ObfuscationHelper.cs (hunk 3 above)
Comment (cristinamanum) [reply src/NuGetGallery/Helpers/ObfuscationHelper.cs]: Updated

File: tests/NuGetGallery.Facts/Helpers/ObfuscationHelperFacts.cs (hunk 4 above)
Comment (cristinamanum) [reply tests/NuGetGallery.Facts/Helpers/ObfuscationHelperFacts.cs]: Updated

**Pull request merged**